### PR TITLE
Feature branch error handling added and dark-ligth mode button bug fixed

### PR DESCRIPTION
**Pull request by OrlandoNicolasMarchioli**

### CODE REVIEW

File: Crypto/Crypto/Presentation/Views/CoinListView/CoinListView.swift [hunk 1]
```diff
@@ -43,6 +48,7 @@ struct CoinListView: View {
         .onReceive(coinListViewModel.$state) { state in
             shouldShowProgressAlert = state.isProgress
             coins = state.cryptoCurrencies
+            fetchError = state.hasError
             // TODO: IMPLEMENT HERE THE ERROR HANDLING IN CASE state.error IS TRUE
```
Comment (axel-endava) [inline Crypto/Crypto/Presentation/Views/CoinListView/CoinListView.swift]: Remove the TODO line

**Review by axel-endava — APPROVED** [review]

File: Crypto/Crypto/Presentation/Views/HomeView/HomeViewModel.swift [hunk 2]
```diff
@@ -12,8 +12,13 @@ class HomeViewModel: ObservableObject{
     @Environment(\.colorScheme) var colorScheme
     @Published var isDarkMode = false
     
-    func toggleDarkMode(){
-        self.isDarkMode.toggle()
+    func toggleMode(mode: String){
+        if(mode == "ligthMode"){
```
Comment (ailinspatino) [inline Crypto/Crypto/Presentation/Views/HomeView/HomeViewModel.swift]: Overall it looks good! Even though there are only two options (dark and light mode) I would suggest handling this with an Enum and a Switch/Case statement, as it is less prone to errors and easier to understand in the code if the states become more complex.

File: Crypto/Crypto/Presentation/Views/LoginView/LogInViewModel.swift [hunk 3]
```diff
@@ -34,7 +34,7 @@ class LogInViewModel: ObservableObject{
                             break
                         case .failure(let error):
                             DispatchQueue.main.async {
-                                self?.state = (self?.state.clone(withIsLoggedIn: false, withError: error.localizedDescription, withAlert: true))!
+                                self?.state = (self?.state.clone(withIsLoggedIn: false,withMessage: "email or password invalid", withAlert: true))!
                             }
```
Comment (ailinspatino) [inline Crypto/Crypto/Presentation/Views/LoginView/LogInViewModel.swift:38]: I would also suggest moving these error messages to a separate file, that way if you need to modify any you have them all in the same place and you don't have to search through the code

File: Crypto/Crypto/Presentation/Views/ProfileView.swift [hunk 4]
```diff
@@ -24,13 +24,14 @@ struct ProfileView: View {
...
+                    .bold()
                 Text("Developer")
                     .padding(.top, 40)
-                    .font(.headline)
+                    .font(.title2)
                 Text("CABA, Buenos Aires, Argentina")
                     .padding(.top, 40)
-                    .font(.headline)
+                    .font(.title2)
                 Spacer()
```
Comment (ailinspatino) [inline Crypto/Crypto/Presentation/Views/ProfileView.swift:36]: Not needed but just a small tip, Enums also come in hand over here. You could use them to handle the different styles and then create a function to apply them to your texts. I've found a [small example](https://medium.com/@paereson/how-to-deal-with-custom-fonts-in-swift-and-swiftui-f7886a824ac8) if you wanna dig into this 😄

**Pull request merged**